### DISCUSSION TITLE
How can setup dcm4chee-art-light development environment on Intellij IDEA?

### DISCUSSION THREAD

**Discussion by Ronaldoshuang** [dcm4che/dcm4chee-arc-light#3452] [Q&A]
Hello everyone!

I am a new with dcm4chee-art-light. Please help me run it on IDE. 

I want to study dcm4chee-art-light ,please help me,thank you very much!

please write to me soon

**Comment by vrindanayak** [comment]
Ensure you have latest maven, java and git installed on your system.
Using _git clone_, clone the [dcm4che](https://github.com/dcm4che/dcm4che.git) library and [dcm4chee-arc-light](https://github.com/dcm4che/dcm4chee-arc-light.git) on to your system. Once your IDE is installed and set up, you may use _File_ -> _New_ -> _Module from existing sources_ and select the cloned dcm4che library. In the following popup window, keep _Import module from external module_ selected with _Maven_ as the selected option. Repeat importing module from existing sources for dcm4chee-arc-light.

**Reply by Ronaldoshuang** [reply]
how to debug source code? i
I took a few steps

1.mvn install -D db=psql
2.add dcm4chee-arc-ear-5.23.2-psql.ear  and dcm4chee-arc-ui2-5.23.2-unsecure.war to  wildfly-25.0.1.Final 
3. Press the button debug

![image](https://user-images.githubusercontent.com/23060328/141438622-99e7b551-f97b-4324-93da-a82caec35bb0.png)

Is that all right?

Can you elaborate?

I want to debug the program in idea because the project is too complex. I want to look at the code step by step and go through the process。

thank you！！

**Reply by vrindanayak** [reply]
By default, the archive is built for psql (Postgres) as DB backend, hence you may just use _mvn install_ instead of _mvn install -D db=psql_.

You may follow [manual Installation](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Installation) or [dockerized Installation](https://github.com/dcm4che/dcm4chee-arc-light/wiki/Run-minimum-set-of-archive-services-on-a-single-host) of archive and enable 8787 port for debugging.

Archive is made up of several modules, so I would suggest to start by selecting some specific module for debugging and understanding. For eg. once you have installed and started the archive in Wildfly, access the archive UI at `http[s]://<host>:<port>/dcm4chee-arc/ui2` which directs you by default to _Navigation_ page _Studies_ tab. 
- As your archive is empty, you may use [storescu](https://github.com/dcm4che/dcm4che/blob/master/dcm4che-tool/dcm4che-tool-storescu/README.md) tool to store some test DICOM studies (ftp://medical.nema.org/MEDICAL/Dicom/DataSets/WG04) to archive and debug the store service present in [dcm4chee-arc-store](https://github.com/dcm4che/dcm4chee-arc-light/tree/master/dcm4chee-arc-store) module 
OR 
- You may click on _Submit_ / _Count_ to debug archive's [DICOM QIDO Restful services](https://petstore.swagger.io/index.html?url=https://raw.githubusercontent.com/dcm4che/dcm4chee-arc-light/master/dcm4chee-arc-ui2/src/swagger/openapi.json#/QIDO-RS) present in [dcm4chee-arc-qido](https://github.com/dcm4che/dcm4chee-arc-light/tree/master/dcm4chee-arc-qido) module.

To debug a particular class / module, set debug points in the respective class / functions. For eg. if you choose to debug [Search for Studies](https://petstore.swagger.io/index.html?url=https://raw.githubusercontent.com/dcm4che/dcm4chee-arc-light/master/dcm4chee-arc-ui2/src/swagger/openapi.json#/QIDO-RS/SearchForStudies) service, set a debug point in [QidoRS.java](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-qido/src/main/java/org/dcm4chee/arc/qido/QidoRS.java) at [searchForStudies](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-qido/src/main/java/org/dcm4chee/arc/qido/QidoRS.java#L239) and continue debugging from there.

Depending upon which [feature / function](https://github.com/dcm4che/dcm4chee-arc-light/wiki/HowTos) you want to use, you may debug the archive modules in different steps. As we also log INFO messages, you may take a look at Wildfly's server logs to understand which archive classes get called for specific functions and debug those modules accordingly.

**Reply by Ronaldoshuang** [reply]
Your help was very much appreciated。
![image](https://user-images.githubusercontent.com/23060328/141601998-748fba58-2507-4cd5-b40a-21abf8b6b56c.png)
The interface returns the types shown in the picture above

but i look at database table,There is no such data type,Is this kind of data a DICOM file read directly

**Reply by vrindanayak** [reply]
On storing studies to archive, certain attribute values are extracted from DICOM files and stored in [study](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-entity/src/main/java/org/dcm4chee/arc/entity/Study.java) / [series](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-entity/src/main/java/org/dcm4chee/arc/entity/Series.java) / [instance](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-entity/src/main/java/org/dcm4chee/arc/entity/Instance.java) tables in DB following the [General Study Module](http://dicom.nema.org/medical/dicom/current/output/html/part03.html#table_C.7-3) / [General Series Module](http://dicom.nema.org/medical/dicom/current/output/html/part03.html#sect_C.7.3.1) / [General Image Module](http://dicom.nema.org/medical/dicom/current/output/html/part03.html#table_C.7-9) respectively; remaining attributes are stored as part of [dicom_attrs](https://github.com/dcm4che/dcm4chee-arc-light/blob/master/dcm4chee-arc-entity/src/main/java/org/dcm4chee/arc/entity/AttributesBlob.java) table. When user / client [queries for studies](https://petstore.swagger.io/index.html?url=https://raw.githubusercontent.com/dcm4che/dcm4chee-arc-light/master/dcm4chee-arc-ui2/src/swagger/openapi.json#/QIDO-RS/SearchForStudies), archive processes the query according to query filters passed in the HTTP request and sends back the attributes according to [Study Resource Search Response Payload](http://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_10.6.3.3.1)

**Reply by Ronaldoshuang** [reply]
How do database fields turn into JSON types? Where is the code?
